### PR TITLE
Add compact proofs API

### PR DESCRIPTION
**Pull request by wemeetagain**
**Description**

- Implement https://github.com/ethereum/beacon-APIs/pull/267

This is a breaking change to our existing getStateProof endpoint.

Before:
```ts
const proof = api.debug.getStateProof(paths);
```

After:
```ts
import {computeDescriptor} from "@chainsafe/persistent-merkle-tree";
import {ssz} from "@lodestar/types";

const gindices = paths.map((path) => ssz.bellatrix.BeaconState.getPathInfo(path).gindex);
const descriptor = computeDescriptor(gindices);
const proof = api.debug.getStateProof(descriptor);
```

### CODE REVIEW

**Comment by wemeetagain** [comment]
Tested by @zsfelfoldi
> Hi! After fixing the format encoding on my side, the new state api seems to work fine 🙂
I updated my "blsync" geth branch that can already be used to test it, and it works fine now:
https://github.com/ethereum/go-ethereum/pull/25901

File: packages/beacon-node/src/api/impl/proof/index.ts [hunk 1]
```diff
@@ -13,29 +14,36 @@ export function getProofApi(
...
+    async getBlockProof(blockId, descriptor) {
+      const {block} = await resolveBlockId(chain.forkChoice, db, blockId);
+
+      // Commit any changes before computing the state root. In normal cases the state should have no changes here
+      const blockNode = config.getForkTypes(block.message.slot).BeaconBlock.toView(block.message).node;
+
+      const data = createProof(blockNode, {type: ProofType.compactMulti, descriptor});
+
+      // descriptor.length / 2 is a rough approximation of # of gindices
+      if (descriptor.length / 2 > maxGindicesInProof) {
```
Comment (twoeths) [inline packages/beacon-node/src/api/impl/proof/index.ts]: @wemeetagain  why this is not the first check in this api? same to `getStateProof`

**Review by twoeths — APPROVED** [review]
looks good to me!

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.6.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.6.0) :tada:

**Pull request merged**